src/results.js

```diff
@@ -31,20 +32,66 @@ class Err {
 	get isErr() { return true }
 	get ok() { return null }
 	get err() { return this._error }
+
 	toString() {
 		if (this._error)
 			return "Error: " + JSON.stringify(this._error)
-		else
-			return "Error"
+		return "Error"
 	}
+
 	toJSON() {
 		let error = (this._error === undefined ? '' : this._error)
 		error = error instanceof Error ? error.toString() : error
 		return { 'Error': error }
 	}
+
+	static _notFound ({ message = 'Not Found', payload, stackTrace } = {}){ 
+		return Err._buildCustomErr('NOT_FOUND', message, payload, stackTrace, 'NotFound', 'notFound')
+	}
+	
+	static _alreadyExists ({ message = 'Entity already exists', payload, stackTrace } = {}) { 
+		return Err._buildCustomErr('ALREADY_EXISTS', message, payload, stackTrace, 'AlreadyExists')
+	}
+	
+	static _invalidEntity ({ message = 'Invalid entity', payload, stackTrace } = {}) { 
```

Comment:
We should have `innerErr` arg for chaining errors.

Ref: https://2ality.com/2021/06/error-cause.html

Reply:
From the list discussed [here](https://github.com/herbsjs/forum/discussions/6#discussioncomment-772007) what would be the behavior for  `Invalid entity` on GraphQL, REST and etc?